substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -1750,6 +1738,13 @@ pub mod pallet {
 	pub type ClaimPermissions<T: Config> =
 		StorageMap<_, Twox64Concat, T::AccountId, ClaimPermission, ValueQuery>;
 
+	/// Records last migration of stake strategy. Stores the migrated strategy type.
+	///
+	/// This is used to ensure that the stake strategy is only migrated once.
+	#[pallet::storage]
+	pub type StrategyMigration<T: Config> =
```

Comment:
Okay I have not seen the migration, but can you elaborate why this is needed if the migration is a one-off MBM? Or is it not?

Reply:
It is single block migration of pools. Currently, there are 229 pools in polkadot and around same or less in Kusama so single block migration seems okay. It is also an unversioned migration (each prod runtime can follow a different migration sequence). 

Probably not super bad but this tries to avoid iterating over these 229 pools again (similar to version checks). Do you suggest to remove it?

Reply:
see my other comment about migration, otherwise this can be closed.